### PR TITLE
DHFPROD-7107: Added mapping parameters example to reference project

### PR DESCRIPTION
**Pull request by rjrudin**
### Description

Also simplified the mapping parameters module interface - renamed argument to be contentSequence so that it's obvious you get a sequence and not an array, and removed stepExecutionContext. The latter isn't available when using the "Test" button in HC, and it's not a public API yet. 

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Review by wooldridge — CHANGES_REQUESTED** [review]
I'm seeing the feature working. The geoPoint field is being populated in the entity instances for the mapCustomersJSON mapping:

Screen Shot 2021-04-30 at 2 26 49 PM

Screen Shot 2021-04-30 at 2 33 35 PM

File: examples/reference-entity-model/README.md
Comment (rjrudin) [reply examples/reference-entity-model/README.md:70]: Changed "then" to "when"

File: examples/reference-entity-model/README.md
Comment (rjrudin) [reply examples/reference-entity-model/README.md:71]: Fixed

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/main.sjs
Comment (srinathgit) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/main.sjs:19]: There should be no "beforeMain" right ? Should there be changes to this file and lib.sjs in this PR ? I will wait until #5594 is merged before I test this out.

**Review by srinathgit — CHANGES_REQUESTED** [review]

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/main.sjs
Comment (rjrudin) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/main.sjs]: Rebased, beforeMain is gone now.

**Review by srinathgit — DISMISSED** [review]

**Comment by rjrudin** [comment]
Rebased

**Review by srinathgit — APPROVED** [review]

**Review by akshaysonvane — APPROVED** [review]

**Review by wooldridge — APPROVED** [review]

**Pull request merged**
